extensions/positron-python/gulpfile.js

```diff
@@ -254,88 +253,98 @@ gulp.task('checkDependencies', gulp.series('checkNativeDependencies'));
 gulp.task('prePublishNonBundle', gulp.series('compile'));
 
 // --- Start Positron ---
-gulp.task('installPythonRequirements', async (done) => {
-    const args = [
+async function pipInstall(args) {
+    await spawnAsync(pythonCommand, [
         '-m',
         'pip',
+
+        // Silence warnings about pip version.
         '--disable-pip-version-check',
+
         'install',
-        '--no-user',
-        '-t',
-        './python_files/lib/python',
+
+        // Upgrade to avoid warnings when rerunning the task.
+        '--upgrade',
+
+        // Args for a safer installation.
         '--no-cache-dir',
-        '--implementation',
-        'py',
         '--no-deps',
-        '--upgrade',
-        '-r',
-        './requirements.txt',
-    ];
-    await spawnAsync(pythonCommand, args)
-        .then(() => true)
-        .catch((ex) => {
-            const msg = "Failed to install requirements using 'python'";
-            fancyLog.error(ansiColors.red(`error`), msg, ex);
-            done(new Error(msg));
-        });
+        '--require-hashes',
+        '--only-binary',
+        ':all:',
 
-    // Vendor Python requirements for the Positron Python kernel.
-    await spawnAsync(pythonCommand, ['scripts/vendor.py']).catch((ex) => {
-        const msg = 'Failed to vendor Python requirements';
-        fancyLog.error(ansiColors.red(`error`), msg, ex);
-        done(new Error(msg));
-    });
-});
+        ...args,
+    ]);
+}
 
-// See https://github.com/microsoft/vscode-python/issues/7136
-gulp.task('installDebugpy', async (done) => {
-    // Install dependencies needed for 'install_debugpy.py'
-    const depsArgs = [
-        '-m',
-        'pip',
-        '--disable-pip-version-check',
-        'install',
-        '--no-user',
-        '--upgrade',
-        '-t',
-        './python_files/lib/temp',
+async function installPythonScriptRequirements() {
```

Comment:
I cleaned this up a bit. I learned that gulp can accept named functions as tasks rather than the `gulp.task('name', ...)` syntax. They can also be async functions, and any errors will be handled appropriately without having to call back to `done`.